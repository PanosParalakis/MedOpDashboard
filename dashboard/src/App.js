import React from 'react';
import './App.css';

import { ajax } from 'rxjs/ajax';
import { combineLatest, of, from } from 'rxjs';
import { delay, mergeMap, repeat, timeoutWith } from 'rxjs/operators';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      temp: null,
      humiditiy: null,
      airPressure: null,
    }

  }

  randomPoll() {
    // 100 - 200ms
    return Math.floor(Math.random() * (200 - 100) + 100);
  }

  getValues(){
    this.observer = combineLatest(
      of({}).pipe(mergeMap(_ => from((ajax.getJSON('http://localhost:4000/temp')).pipe(delay(this.randomPoll()), timeoutWith(1000, [{"value": "N/A"}])))),repeat()),
      of({}).pipe(mergeMap(_ => from((ajax.getJSON('http://localhost:4000/humidity')).pipe(delay(this.randomPoll()), timeoutWith(1000, [{"value": "N/A"}]) ))),repeat()),
      of({}).pipe(mergeMap(_ => from((ajax.getJSON('http://localhost:4000/air-pressure')).pipe(delay(2000), timeoutWith(1000, [{"value": "N/A"}])))),repeat()),
    ).subscribe(res => this.setState({
      temp: res[0].value,
      humiditiy: res[1].value,
      airPressure: res[2].value,
    }));
  }


  componentDidMount() {
    this.getValues();
  }

  componentWillUnmount() {
    this.observer.unsubscribe();
  }

  render() {
    return (
      <div>
        <div id="temp">Temp (C): <span>{this.state.temp}</span></div>
        <div id="humiditiy">Humiditiy (%): <span>{this.state.humiditiy}</span></div>
        <div id="airPressure">Air pressure (psi): <span>{this.state.airPressure}</span></div>
      </div>
    );
  }
}

export default App;
