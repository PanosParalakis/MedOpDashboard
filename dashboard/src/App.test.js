import React from 'react';
// setup file
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import App from './App';

import { ajax } from 'rxjs/ajax';
import { combineLatest, of, from } from 'rxjs';
import { delay, mergeMap, repeat, timeoutWith } from 'rxjs/operators';

configure({ adapter: new Adapter() });
let container = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("should render app", () => {
  act(() => {
    render(<App />, container);
  });

  expect(container.textContent).toBe("Temp (C): Humiditiy (%): Air pressure (psi): ");
});


test("it should match snapshot", done => {
  const component = shallow(<App />);

  expect(component.getElements()).toMatchSnapshot();
  done();
});
