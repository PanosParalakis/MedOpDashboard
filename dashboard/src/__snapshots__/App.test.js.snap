// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot 1`] = `
Array [
  <div>
    <div
      id="temp"
    >
      Temp (C): 
      <span />
    </div>
    <div
      id="humiditiy"
    >
      Humiditiy (%): 
      <span />
    </div>
    <div
      id="airPressure"
    >
      Air pressure (psi): 
      <span />
    </div>
  </div>,
]
`;
